src/registry/customerProfile/subFields.tsx

```diff
@@ -0,0 +1,332 @@
+
+const DigitalAddressesSelect: React.FC<DigitalAddressesSelectProps> = ({
+  type,
+  fieldName,
+  isRequired,
+  digitalAddressGroup,
+  onAddDigitalAddress,
+}) => {
+  const intl = useIntl();
+  const options = getDigitalAddressOptions(intl, digitalAddressGroup);
+  const formSettings = useFormSettings();
+  if (!formSettings?.componentParameters?.customerProfile) {
+    throw new Error('Customer profile component parameters not configured');
+  }
+
+  const {portalUrl} = formSettings.componentParameters.customerProfile;
+  return (
+    <>
+      <Select
+        name={fieldName}
+        label={<FormattedMessage {...FIELD_LABELS[type]} />}
+        options={options}
+        isRequired={isRequired}
+        optionComponent={OptionWithDescription}
+        isDisabled={options.length === 1}
+      />
+      <Alert type="info">
+        <FormattedMessage
+          description="Profile digital address multiple addresses: edit preference text"
+          defaultMessage={`There are multiple {digitalAddressType, select,
+              email {email addresses}
+              phoneNumber {phone numbers}
+              other {{digitalAddressType}}
+            } associated with your account. Use the dropdown to select the {digitalAddressType, select,
+              email {email address}
+              phoneNumber {phone number}
+              other {{digitalAddressType}}
+            } you want to use for this form. To update your preferences, use the online <a>portal</a>.
+          `}
+          values={{
+            digitalAddressType: type,
+            a: (...chunks) => (
+              <a href={portalUrl} target="_blank" rel="noopener noreferrer">
+                {chunks}
+              </a>
+            ),
+          }}
+        />
+      </Alert>
+      <ButtonGroup>
+        <SecondaryActionButton onClick={onAddDigitalAddress}>
+          <FormattedMessage
+            description="Profile digital address 'add digital address' button label"
+            defaultMessage={`Add {digitalAddressType, select,
+              email {email address}
+              phoneNumber {phone number}
+              other {{digitalAddressType}}
+            }`}
+            values={{digitalAddressType: type}}
+          />
+        </SecondaryActionButton>
+      </ButtonGroup>
+    </>
+  );
+};
+
+interface DigitalAddressTextfieldProps extends DigitalAddressSubFieldProps {
+  namePrefix: string;
+  textfieldProps: Partial<React.ComponentProps<typeof TextField>>;
+}
+
+const DigitalAddressTextfield: React.FC<DigitalAddressTextfieldProps> = ({
+  type,
+  namePrefix,
+  fieldName,
+  isRequired,
+  textfieldProps,
+}) => {
+  const {setFieldValue, getFieldMeta} = useFormikContext();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const {value, touched, error} = getFieldMeta(fieldName);
+
+  // Only show the preference button if the field is touched, has a value and has no error
+  const showPreferencesButton = touched && !error && value !== '';
```

Comment:
this negates part of my earlier remark about setting touched state when this button is clicked - I think this behaviour makes most sense